newm-server/src/main/kotlin/io/newm/server/features/song/repo/SongRepositoryImpl.kt

```diff
@@ -104,21 +106,27 @@ internal class SongRepositoryImpl(
         }
     }
 
-    override suspend fun generateUploadUrl(songId: UUID, requesterId: UUID, fileName: String): String {
-        logger.debug(marker, "generateUploadUrl: songId = $songId, fileName = $fileName")
+    override suspend fun generateUploadUrl(
+        uploadType: UploadType,
+        songId: UUID,
+        requesterId: UUID,
+        fileName: String
+    ): String {
+        logger.debug(marker, "generateUploadUrl: uploadType = $uploadType, songId = $songId, fileName = $fileName")
```

Comment:
for debug messages that contain a lot of string concatenation, surround them with 

```
if(logger.isDebugEnabled) { ... }
```

so that work can be skipped in most cases.

Reply:
That's an excellent point. It got me thinking into creating a `Logger` extension function that combines both the `logger.isDebugEnabled` and `logger.debug` with a lambda parameter that returns the concatenated string. That way the lambda (which does the concats) only gets when debug enabled. I'll do some more brainstorming and create separate PR.